drivers/pwm/pwm_stm32.c

```diff
@@ -125,43 +125,135 @@ static int get_tim_clk(const struct stm32_pclken *pclken, uint32_t *tim_clk)
+	 *   frequency of the APB domain.
+	 *
+	 * Ref. RM0402 Rev. 5, p. 160 (F412)
+	 * Ref. RM0430 Rev. 8, p. 177 (F413, F423)
+	 * Ref. RM0090 Rev. 18, p. 208 (F427, F437, F429, F439)
+	 * Ref. RM0390 Rev. 4, p 168 (F446)
+	 * Ref. RM0386 Rev. 5, p. 191 (F469, F479)
+	 *
+	 * F72X, F73X, F74X, F75X, F76X, F77X:
+	 *
+	 * - TIMPRE=0: If the APB{1,2} prescaler is configured to a division
+	 *   factor of 1 then the timer clock equals to HCLK (which is equal to
+	 *   the APB bus clock in this case). Otherwise the timer clock is set
+	 *   to twice the frequency of the APB bus clock. In case prescaler is
+	 *   2 it also results in HCLK.
+	 * - TIMPRE=1: If the APB{1,2} prescaler is configured to a division
+	 *   factor of 1, 2 or 4, then the timer clock equals to HCLK.
+	 *   Otherwise, the timer clock frequencies are set to four times to the
+	 *   frequency of the APB domain.
+	 *
+	 * Ref. RM0431 Rev. 3, p. 186 (F72X, F73X)
+	 * Ref. RM0385 Rev. 8, p. 192 (F74X, F75X)
+	 * Ref. RM0410 Rev. 4, p. 213 (F76X, F77X)
+	 *
+	 * H723, H733, H725, H735, H730, H742, H743, H753, H750, H745, H755,
+	 * H747, H757:
+	 *
+	 * - TIMPRE=0: If the APB{1,2} prescaler is configured to a division
+	 *   factor of 1 or 2 then the timer clock equals to HCLK. Otherwise the
+	 *   timer clock is set to twice the frequency of APB bus clock.
+	 * - TIMPRE=1: If the APB{1,2} prescaler is configured to a division
+	 *   factor of 1, 2 or 4, then the timer clock equals to HCLK.
+	 *   Otherwise, the timer clock frequencies are set to four times to the
+	 *   frequency of the APB domain.
+	 *
+	 * Ref. RM0468 Rev. 2, p. 369 (H723, H733, H725, H735, H730)
+	 * Ref. RM0433 Rev. 7, p. 388 (H742, H743, H753, H750)
+	 * Ref. RM0399 Rev. 3, p. 423 (H745, H755, H747, H757)
+
+	 * H7A3, H7B3, H7B0:
+	 *
+	 * - TIMPRE=0: If the APB{1,2} prescaler is configured to a division
+	 *   factor of 1 or 2 then the timer clock equals to HCLK. Otherwise the
+	 *   timer clock is set to twice the frequency of APB bus clock.
+	 * - TIMPRE=1: If the APB{1,2} prescaler is configured to a division
+	 *   factor of 1, 2 or 4, then the timer clock equals to twice the
+	 *   frequency of the APB bus clock. Otherwise, the timer clock
+	 *   frequencies are set to four times to the frequency of the APB
+	 *   domain.
+	 *
+	 * Ref. RM0455 Rev. 4, p. 390 (H7A3, H7B3, H7B0)
 	 */
+
 	if (LL_RCC_GetTIMPrescaler() == LL_RCC_TIM_PRESCALER_TWICE) {
-		if (apb_psc == 1u) {
-			*tim_clk = bus_clk;
+		/* TIMPRE = 0 */
+		if (apb_psc == 1u || apb_psc == 2u) {
+			LL_RCC_ClocksTypeDef clocks;
+
+			LL_RCC_GetSystemClocksFreq(&clocks);
+			*tim_clk = clocks.HCLK_Frequency;
 		} else {
 			*tim_clk = bus_clk * 2u;
 		}
 	} else {
+		/* TIMPRE = 1 */
+#if defined(CONFIG_SOC_STM32F401XC) || defined(CONFIG_SOC_STM32F401XE) ||      \
+	defined(CONFIG_SOC_STM32F411XE)
+		if (apb_psc == 1u || apb_psc == 2u) {
+#else
 		if (apb_psc == 1u || apb_psc == 2u || apb_psc == 4u) {
-			*tim_clk = SystemCoreClock;
+#endif
```

Comment:
This can be simplified further:
you can get ride of #if
because for apb_psc == 4u --> HCLK==bus_clk * 4u
You can even choose between 
`if (apb_psc == 1u || apb_psc == 2u) {`
or 
`if (apb_psc == 1u || apb_psc == 2u || apb_psc == 4u) {`

But I would prefer the 2nd one to be coherent with TIMPRE = 0